### PR TITLE
Fix broken interop test, using new StreamObserver API.

### PR DESCRIPTION
**Pull request by zsurocking**

### CODE REVIEW

File: android-interop-testing/app/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ android {
 
     defaultConfig {
         applicationId "io.grpc.android.integrationtest"
-        minSdkVersion 9
+        minSdkVersion 19
```
Comment (madongfly) [inline android-interop-testing/app/build.gradle]: Why do we need this change?

Comment (zsurocking) [inline android-interop-testing/app/build.gradle]: Already reverted.

**Comment by madongfly** [comment]
@zsurocking LGTM

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @zsurocking isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Pull request merged**